ampd/src/lib.rs

```diff
@@ -275,7 +281,34 @@ where
         };
 
         while (set.join_next().await).is_some() {}
+        let state_update_result = state_rx.try_recv().change_context(Error::ReturnState);
 
```

Comment:
`try_recv` may return `Err(Empty)` if no pending value exists in the channel at the moment. How can we be sure that the state_updaer task has exited at this moment? It may be still waiting for the handlers to terminate first.